File: kennel-viewer/src/app/models/Pet.ts
import { TimeRecord } from "./TimeRecord";
import { EmergencyContact } from "./EmergencyContact";
import { Client } from "./Client";

export interface Pet{
    id: number; //Find some way to generate the value
    name: string;
    type: string; // Potential enum, default to dog normally
    breed: string; 
    age: number;
    owners: Client[];
    emergencyContacts: EmergencyContact[];
    checkIns: TimeRecord[];
    violations: number;
}

File: kennel-manager/src/main/java/com/springsrescuemission/kennelmanager/model/Pet.java
package com.springsrescuemission.kennelmanager.model;

import lombok.Data;

import java.time.LocalDate;
import java.util.Map;
import java.util.Set;

@Data
public class Pet {

    private String name;
    private String type; //TODO what will enums look like?
    private String breed;
    private Integer age;
    private Set<Owner> Owners;
    private Set<EmergencyContact> emergencyContacts;
    private Map<LocalDate, TimeRecord> checkIns;
    private Integer violations;

}
